src/main/java/com/celements/nextfreedoc/NextFreeDocService.java

```diff
@@ -61,11 +67,9 @@ private long getNextTitledPageNum(DocumentReference baseDocRef) {
 
   private boolean isAvailableDocRef(DocumentReference newDocRef) {
     try {
-      return (!getContext().getWiki().exists(newDocRef, getContext())
-          && (getContext().getWiki().getDocument(newDocRef, getContext()).getLock(
-              getContext()) == null));
-    } catch (XWikiException exp) {
-      LOGGER.info("Failed to check new document reference [" + newDocRef + "].", exp);
+      return (modelAccess.createDocument(newDocRef).getLock(getContext()) == null);
+    } catch (XWikiException | DocumentLoadException | DocumentAlreadyExistsException exp) {
+      LOGGER.info("Failed to check new document reference [{}].", newDocRef, exp);
```

Comment:
use different logging levels for exceptions:
DocumentAlreadyExistsException: frequently expected, trace or debug
XWikiException | DocumentLoadException: unexpected, error